### PR TITLE
:sparkle: pkg/internal/codegen/parse: "Title" CRD validation annotation

### PR DESCRIPTION
**Pull request by estroz**
This PR implements the [`SchemaProps.Title`](https://godoc.org/github.com/go-openapi/spec#SchemaProps) field as an annotation: `+kubebuilder:validation:Title="some title"`, where the quoted string following `=` is stripped of surrounding quotes and interpreted literally as a field's title. This validation is not inherited by child props.

Reference: https://github.com/kubernetes-sigs/controller-tools/issues/156#issuecomment-479632914

/cc @mengqiy @DirectXMan12 @droot

### CODE REVIEW

File: pkg/internal/codegen/parse/crd.go [hunk 1]
```diff
@@ -577,6 +581,20 @@ func getValidation(comment string, props *v1beta1.JSONSchemaProps) {
 		}
 	case "Format":
 		props.Format = parts[1]
+	case "Title":
+		title := strings.TrimSpace(strings.Join(parts[1:], " "))
+		if len(title) > 2 {
+			// Remove beginning and ending quote from non-empty strings.
+			if title[0] == '"' {
```
Comment (DirectXMan12) [inline pkg/internal/codegen/parse/crd.go]: this accepts `Title=foo"`, which doesn't seem to be valid.  Also, `Title=f` should probably be valid.  I think we may need to hold on here a bit and define more format parsing semantics for our markers (I've got some stuff in the works that I'll post soon).

Comment (estroz) [reply pkg/internal/codegen/parse/crd.go]: We could simply require string fields be wrapped in `""` and anything inside will be interpreted literally.

Comment (DirectXMan12) [reply pkg/internal/codegen/parse/crd.go]: I've got parsing logic written.  We should be consistent across our different annotations.

Comment (estroz) [reply pkg/internal/codegen/parse/crd.go]: Has it been merged?

Comment (DirectXMan12) [reply pkg/internal/codegen/parse/crd.go]: not yet, it's part of a broader overhaul being tracked on the crdgenerator feature branch

**Comment by DirectXMan12** [comment]
this is integrated into v0.2.0, so it should be all set now.

**Pull request closed without merging**